### DISCUSSION TITLE
Could not find main module for @johnlindquist/kit

### DISCUSSION THREAD

**Discussion by AquiGorka** [johnlindquist/kit#1389] [Errors]
Error:
```
Error: Could not find main module for @johnlindquist/kit
Error: Error: Could not find main module for @johnlindquist/kit
    at kenvImport (file:///Users/Fynn/.kit/api/npm.js:83:15)
    at async file:///Users/Fynn/.kit/cli/error-action.js
```

Screenshot:
![Screenshot 2023-12-04 at 14 40 49](https://github.com/johnlindquist/kit/assets/3072458/af2395c2-26f9-4c01-a365-b37dd91db5df)

**Comment by AquiGorka** [comment]
I have installed `kit` in another user in the same laptop and everything works correctly, would that have something to do with this error?

**Reply by johnlindquist** [reply]
Unfortunately, this is a bug with the error handler:
1. An error fires
2. The Kit error handler catches the error
3. The Kit error handler fails to parse the error
4. The error handler catches _that_ error then displays the error you're seeing (that's why it looks like a strange internal error)

To find the "real" error:
1. Open ~/.kit/api/kit.js
2. Find the line: `export let errorHandler`
3. Add a `log({error})` as the first line in the function.
4. Run your script 2x (the first time will probably have a cached version of that file)
5. Check `~/.kit/logs/kit.log` for the error

The solution is almost always a misbehaving package:
1. cd ~/.kenv
2. rm -rf ./node_modules
3. ~/.knode/bin/npm i

Let me know if that helps you find it
